libnmstate/nm/ipv4.py

```diff
@@ -60,8 +67,23 @@ def get_info(active_connection):
     if active_connection is None:
         return info
 
+    set_ip_conf = _get_setting_ip_conf(active_connection)
+    if set_ip_conf:
+        info['dhcp'] = set_ip_conf.get_method() == (
+            nmclient.NM.SETTING_IP4_CONFIG_METHOD_AUTO)
+    else:
+        info['dhcp'] = False
+
     ip4config = active_connection.get_ip4_config()
     if ip4config is None:
+        # When DHCP is enable, it might be possible, the active_connection does
```

Comment:
I hope that active_connection.get_dhcp4_config() should be enough to determine that IPv4 is enabled.
Then both can set the `enabled` value.

Reply:
`active_connection.get_dhcp4_config()` is getting the succeeded DHCP information.
For non-going DHCP or failed DHCP, you get `None`. We still need to check profile for that.